Tasks/AzureCloudPowerShellDeploymentV1/Utility.ps1

```diff
@@ -53,7 +53,23 @@ function Get-RoleName($extPath)
     return $roleName
 }
 
-function Get-DiagnosticsExtensions($storageAccount, $extensionsPath, $storageAccountKeysMap)
+function Get-AzureStoragePrimaryKey($storageAccount, [bool]$isArm)
+{
+    if ($isArm)
+    {
+        $storageAccountResource = Get-AzureRmResource -Name $storageAccount -ResourceType "Microsoft.Storage/storageAccounts"
+        if (!$storageAccountResource)
+        {
+            Write-Error -Message "Could not find resource $storageAccount that has a type of Microsoft.Storage/storageAccounts"
+        }
+        $primaryStorageKey = (Get-AzureRmStorageAccountKey -ResourceGroupName $storageAccountResource.ResourceGroupName -Name $storageAccount)[0].value
```

Comment:
Can we have an additional check over the length of array before accessing index 0